### ISSUE TITLE
Namespace created from exports reporting that the exports are unused when using namespace imports

### ISSUE THREAD

**Issue by shawnmcknight** [webpro-nl/knip#690]
I've created a reproduction at https://github.com/shawnmcknight/knip-import-exported-ns.

The scenario I am seeing is that when a namespace is imported via `import * as ns` and that namespace is created via re-exporting other modules then there is a false positive on those re-exported modules being unused exports.

In my example reproduction, there are three subfolders as follows:
- `ns-not-reexported`: This folder contains an `index.ts` file which directly exports two named exports - `apple` and `orange`.
- `ns-named-reexported`: This folder contains files named `apricot.ts` and `peach.ts` which individually export named `apricot` and `peach`. These are then re-exported as named exports in the folder's `index.ts`.
- `ns-default-reexported`: This folder contains files named `blueberry.ts` and `pear.ts` which export default `blueberry` and `pear` respectively. These default exports and then re-exported as named exports in the folder's `index.ts`.

In the root folder, there is an `index.ts` which imports each of these three subfolders using `import * as ns` syntax and then simply does a `console.log` of each of those. Based upon my understanding of the [namespace import heuristic](https://knip.dev/guides/namespace-imports#the-default-heuristic), this should result in no unused exports during knip's validation.

The namespace in the `ns-not-reexported` folder is _not_ reported as an unused export. However, the namespaces in the `ns-default-reexported` and `ns-named-reexported` folders are reported as unused exports. It seems that the re-exporting process is causing those to be missed.

Here is the knip output from my reproduction:
```
Unused exports (4)
default  unknown  ns-default-reexported/blueberry.ts:2:8
default  unknown  ns-default-reexported/pear.ts:2:8
apricot  unknown  ns-named-reexported/apricot.ts:1:14
peach    unknown  ns-named-reexported/peach.ts:1:14
```

The `apple` and `orange` exports are properly identified as being used, but the other four are being flagged as unused.

Please let me know if there is anything else I can provide to assist!

**Comment by shawnmcknight** [comment]
This is possibly related to #626, but I'm not totally sure.

**Comment by webpro** [comment]
Thanks for reproduction, looks clear. There's quite a few cases covered in `re-exports-*` fixture folders, but apparently this one is missing indeed. To be continued.

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v5.22.3. See [Release 5.22.3](https://github.com/webpro-nl/knip/releases/tag/5.22.3) for release notes._

_Using Knip in a commercial project? Please consider [becoming a sponsor](https://knip.dev/sponsors)._

**Comment by shawnmcknight** [comment]
I just tested in my repo and it's working as expected now. Thank you!
